File: Backend/src/main/java/com/university/elecitvecoursesappmanagement/service/EnrollmentService.java
package com.university.elecitvecoursesappmanagement.service;

import com.university.elecitvecoursesappmanagement.dto.EnrollmentDTO;
import com.university.elecitvecoursesappmanagement.entity.Enrollment;

import java.util.List;
import java.util.Optional;

public interface EnrollmentService {
    List<EnrollmentDTO> getAllEnrollments();

    Optional<EnrollmentDTO> getEnrollmentById(Long id);

    Enrollment addEnrollment(Enrollment enrollment);

    Enrollment updateEnrollment(Long id, Enrollment updateEnrollment);

    void deleteEnrollmentById(Long id);

    void deleteAllEnrollments();
}

File: Frontend/elective-course-app-frontend/src/app/enrollment.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from "@angular/common/http";
import { Observable } from "rxjs";
import { Enrollment } from "./enrollment";

@Injectable({
  providedIn: 'root'
})
export class EnrollmentService {

  private baseURL = "http://localhost:8080/api/enrollment";
  private getURL = `${this.baseURL}/getAllEnrollments`;
  private getByIdURL = `${this.baseURL}/getEnrollmentById`;
  private createURL = `${this.baseURL}/addEnrollment`;
  private updateURL = `${this.baseURL}/updateEnrollment`;
  private deleteURL= `${this.baseURL}/deleteEnrollmentById`;

  constructor(private httpClient: HttpClient) { }

  getEnrollmentList(): Observable<Enrollment[]> {
    return this.httpClient.get<Enrollment[]>(this.getURL);
  }

  createEnrollment(enrollment: Enrollment): Observable<Enrollment> {
    return this.httpClient.post<Enrollment>(this.createURL, enrollment);
  }
  

  getEnrollmentById(id: number | undefined): Observable<Enrollment> {
    return this.httpClient.get<Enrollment>(`${this.getByIdURL}/${id}`);
  }

  updateEnrollment(id: number | undefined, enrollment: Enrollment): Observable<Enrollment> {
    return this.httpClient.put<Enrollment>(`${this.updateURL}/${id}`, enrollment);
  }

  deleteEnrollment(id: number | undefined): Observable<void> {
    return this.httpClient.delete<void>(`${this.deleteURL}/${id}`);
  }
}
